packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m

```diff
@@ -389,9 +394,16 @@ - (BOOL)tryCopyImage:(NSDictionary * _Nonnull)info path:(NSString *)toPath {
   return false;
 }
 
-- (void)handleVideoWithInfo:(NSDictionary * _Nonnull)info saveAt:(NSString *)directory updateResponse:(NSMutableDictionary *)response
+- (void)handleVideoWithInfo:(NSDictionary * _Nonnull)info
+                     saveAt:(NSString *)directory
+             updateResponse:(NSMutableDictionary *)response
+          completionHandler:(void (^)(void))completionHandler
 {
-  NSURL *videoURL = info[UIImagePickerControllerMediaURL];
+  NSURL *videoURL = info[UIImagePickerControllerMediaURL] ? info[UIImagePickerControllerMediaURL] : info[UIImagePickerControllerReferenceURL];
+  if (videoURL == nil) {
+    self.reject(@"E_COULDNT_OPEN_FILE", @"Couldn't open video", nil);
+    return;
```

Comment:
Ok, I see that the answer is **no**, but couldn't we change the logic around this piece of code so that we always call `completionHandler` before returning from this function? We could pass an updated `response` back to the handler in case of success and `nil` in case of failure.

1. We wouldn't need to remember `self.reject` as an instance variable.
2. We would always call `completionHandler` which is something someone would expect from a completion handler
3. resolving/rejecting the promise is in one place

Reply:
I think it's a good idea. However, I'll do it in different PR. It's not connected with export presets and this change will affect others part of this module.